source/AsyncRgbLedAnalyzerSettings.cpp

```diff
@@ -67,7 +107,46 @@ const char* AsyncRgbLedAnalyzerSettings::SaveSettings()
 	SimpleArchive text_archive;
 
 	text_archive << mInputChannel;
-	text_archive << mBitRate;
+	text_archive << mLEDController;
 
 	return SetReturnString( text_archive.GetString() );
 }
+
+U8 AsyncRgbLedAnalyzerSettings::BitSize() const
+{
+    return mControllers.at(mLEDController).mBitsPerChannel;
+}
+
+U8 AsyncRgbLedAnalyzerSettings::LEDChannelCount() const
+{
+    return mControllers.at(mLEDController).mChannelCount;
+}
+
+U32 AsyncRgbLedAnalyzerSettings::DataTimeNSecHigh(BitState value) const
+{
+    const auto& d = mControllers.at(mLEDController);
+    return mIsHighSpeedMode ? d.mDataTimingHighSpeedNsec[value][0]
+                            : d.mDataTimingNSec[value][0];
+}
+
+U32 AsyncRgbLedAnalyzerSettings::DataTimeNSecLow(BitState value) const
+{
+    const auto& d = mControllers.at(mLEDController);
+    return mIsHighSpeedMode ? d.mDataTimingHighSpeedNsec[value][1]
+                            : d.mDataTimingNSec[value][1];
+}
+
+bool AsyncRgbLedAnalyzerSettings::IsHighSpeedSupported() const
+    {
+    return mControllers.at(mLEDController).mHasHighSpeed;
+}
+
+U32 AsyncRgbLedAnalyzerSettings::ResetTimeNSec() const
+{
+    return mControllers.at(mLEDController).mResetTimeNSec;
+}
+
+auto AsyncRgbLedAnalyzerSettings::GetColorLayout() const -> ColorLayout
```

Comment:
This syntax is useful for class-scoped enums, but since I moved ColorLayout into the Helpers.h, we no longer even need this syntax, so I can remove it.